File: Raspberry Pi/Sense Hat Examples/sensors_via_deamon.py
#!/usr/bin/python
import cgi
#import cgitb; cgitb.enable()  # for troubleshooting
import socket
import json

def getresource(id):
    # create a TCP/IP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    # connect the socket to the 'sense_hat_service' port: 10000
    server_address = ('localhost', 10000)
    sock.connect(server_address)
    res = '[null]'
    try:
        # send command
        cmd = 'get_' + id
        cmd = cmd.encode('utf-8') 
        sock.sendall(cmd)
        # receive resource
        res = sock.recv(256)
        res = res.decode('utf-8')
    finally:
        sock.close()
    return res

print("Content-Type: application/json\n\n")
# script input arguments
arg = cgi.FieldStorage()

if 'id' in arg.keys():
    # resources IDs
    ids = ['env', 'rpy']
    # check selected ID
    id = [id for id in ids if id == arg['id'].value]
    if id:
        print(getresource(id[0]))
    else:
        print('[]')
else:
    print('[]')

File: Raspberry Pi/Sense Hat Examples/sensors_via_deamon.php
<?php
error_reporting(E_ALL);

function getresource($id) {
    // create a TCP/IP socket
    $socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
    if ($socket === false) {
        echo "socket_create() failed: reason: " . socket_strerror(socket_last_error()) . "\n";
    }
    // connect the socket to the 'sense_hat_service' port: 10000
    $result = socket_connect($socket, 'localhost', 10000);
    if ($result === false) {
        echo "socket_connect() failed.\nReason: ($result) " . socket_strerror(socket_last_error($socket)) . "\n";
    }
    // send command
    $cmd = "get_" . $id;
    socket_write($socket, $cmd, strlen($cmd));
    // receive resource
    $res = socket_read($socket, 256);
    socket_close($socket);
    return $res;
}

header('Content-Type: application/json');

if(isset($_GET['id'])) {
    // resources IDs
    $ids = array('env', 'rpy');
    // check selected ID
    $idno = array_search($_GET['id'], $ids);
    if($idno === false) {
        echo '[]';
    }
    else { 
        $id = $ids[$idno];
        echo getresource($id);
    }
}
else {
    echo '[]';
}
?>